Source/core/Sync.cpp

```diff
@@ -782,6 +784,224 @@ namespace Core {
         return (nResult);
     }
 
+    //----------------------------------------------------------------------------
+    //----------------------------------------------------------------------------
+    // SharedSemaphore class
+    //----------------------------------------------------------------------------
+    //----------------------------------------------------------------------------
+
+    //----------------------------------------------------------------------------
+    // CONSTRUCTOR & DESTRUCTOR
+    //----------------------------------------------------------------------------
+
+
+    SharedSemaphore::SharedSemaphore(const TCHAR sourceName[], const uint32_t initCount, VARIABLE_IS_NOT_USED const uint32_t maxCount)
+    {
+        ASSERT(initCount <= 1);
+        ASSERT(maxCount == 1);
+
+#ifdef __WINDOWS__
+        _semaphore = (::CreateSemaphore(nullptr, initCount, maxCount, sourceName));
+        ASSERT(_semaphore != nullptr);
+#else
+        _name = "/" + string(sourceName);
+        _semaphore = sem_open(_name.c_str(), O_CREAT | O_RDWR, 0644,  initCount);
```

Comment:
I think it also needs O_EXCL, so we block accidental reuse of the semaphore with this constructor dedicated to creating